synapse/rest/admin/users.py

```diff
@@ -725,12 +725,10 @@ def __init__(self, hs):
     async def on_GET(self, request, user_id):
         await assert_requester_is_admin(self.auth, request)
 
-        if not self.is_mine(UserID.from_string(user_id)):
-            raise SynapseError(400, "Can only lookup local users")
-
-        user = await self.store.get_user_by_id(user_id)
-        if user is None:
-            raise NotFoundError("Unknown user")
+        if self.is_mine(UserID.from_string(user_id)):
+            # Let's get rid of unknown local users quickly.
+            if await self.store.get_user_by_id(user_id) is None:
+                raise NotFoundError("Unknown user")
```

Comment:
> Is this an optimization?

It looks like an optimization. I had to remove it from the case of non-local users because it always fired in that case.

> Should we also raise this for remote users if no rooms are found?

Not sure. What do you think? I don't see a benefit to raising `NotFoundError` vs. returning an empty list.

Reply:
I think it might be confusing to have different behavior between local and remote users, but at the very least we should document this in the API docs.

Reply:
I believe that the least confusing behavior would then be to entirely remove the `NotFoundError`, what do you think?